src/launchpad/analyzers/range_mapping_builder.py

```diff
@@ -0,0 +1,485 @@
+                            current_pos,
+                            range_item.start,
+                            BinaryTag.DATA_SEGMENT,
+                            f"linkedit_large_gap_{gap_size}_bytes",
+                            allow_partial=True,
+                        )
+
+                current_pos = max(current_pos, range_item.end)
+
+            # Fill any remaining gap at the end
+            if current_pos < linkedit_end:
+                gap_size = linkedit_end - current_pos
+
+                if gap_size <= 1024:
+                    logger.debug(f"Filling end padding {gap_size} bytes at offset {current_pos}")
+                    range_map.add_range(
+                        current_pos,
+                        linkedit_end,
+                        BinaryTag.UNMAPPED,
+                        f"linkedit_end_padding_{gap_size}_bytes",
+                        allow_partial=True,
+                    )
+                elif gap_size <= 65536:
+                    logger.debug(f"Filling end data {gap_size} bytes at offset {current_pos}")
+                    range_map.add_range(
+                        current_pos,
+                        linkedit_end,
+                        BinaryTag.DATA_SEGMENT,
+                        f"linkedit_end_data_{gap_size}_bytes",
+                        allow_partial=True,
+                    )
+                else:
+                    logger.warning(f"Large LINKEDIT end gap: {gap_size} bytes at offset {current_pos}")
+                    range_map.add_range(
+                        current_pos,
+                        linkedit_end,
+                        BinaryTag.DATA_SEGMENT,
+                        f"linkedit_end_large_gap_{gap_size}_bytes",
+                        allow_partial=True,
+                    )
+
+        except Exception as e:
+            logger.debug(f"Failed to map linkedit gaps: {e}")
+
+    def _map_remaining_gaps(self, range_map: RangeMap) -> None:
+        """Map any remaining unmapped regions as padding/alignment."""
+        try:
+            unmapped_regions = range_map.get_unmapped_regions()
+
+            for region in unmapped_regions:
+                # Only map gaps smaller than 64KB as padding/alignment
+                # Larger gaps might indicate missing structures that need specific mapping
+                if region.size <= 65536:  # 64KB threshold
+                    range_map.add_range(
+                        region.start,
+                        region.end,
+                        BinaryTag.UNMAPPED,
+                        f"padding_gap_{region.size}_bytes",
+                        allow_partial=True,  # Use partial splitting for padding gaps
+                    )
+                else:
+                    logger.warning(f"Large unmapped region: {region.size} bytes at offset {region.start}")
+        except Exception as e:
+            logger.debug(f"Failed to map remaining gaps: {e}")
```

Comment:
honestly didn't look too closely at this file since it's so bespoke and just ripped from our existing code. Will just defer to shadow / manual testing for reliability of this logic